source/extensions/tracers/xray/tracer.cc

```diff
@@ -76,17 +77,28 @@ void Span::finishSpan() {
   s.set_end_time(
       time_point_cast<SecondsWithFraction>(time_source_.systemTime()).time_since_epoch().count());
   s.set_parent_id(parentId());
-  using KeyValue = Protobuf::Map<std::string, std::string>::value_type;
-  for (const auto& item : custom_annotations_) {
-    s.mutable_annotations()->insert(KeyValue{item.first, item.second});
+
+  // HTTP annotations
+  using StructField = Protobuf::MapPair<std::string, ProtobufWkt::Value>;
+  daemon::Segment_http_annotations* http = s.mutable_http();
+
+  ProtobufWkt::Struct* request = http->mutable_request();
+  auto request_fields = request->mutable_fields();
+  for (const auto& field : http_request_annotations_) {
+    request_fields->insert(StructField{field.first, field.second});
+
+    (*request_fields)[field.first] = field.second;
```

Comment:
This line is a duplicate. remove?

Reply:
bleh, looking at this and cout it looks like I didn't look at my rebase merge closely enough. fixed.